docs/UserGuide.md

```diff
@@ -2,158 +2,474 @@
 layout: page
 title: User Guide
 ---
+# EduMate User Guide
+
+EduMate is a desktop app designed for NUS students to manage their academic and social lives.
+
+## Using this Guide
+If this is the first time you are using this user guide, we highly recommend you to read the [Overview](#overview) section. Otherwise,
+
+* If you are running EduMate for the first time, please take a look at our [Quick Start](#quick-start) guide.
+* If you want to learn to use EduMate, do check out our [Commands](#) section for a detailed guide.
+* If you want to contribute to this project, please refer to our [Developer Guide](https://ay2223s2-cs2103t-w14-2.github.io/tp/DeveloperGuide.html).
+
+## Table of Contents
+1. [**Using this Guide**](#using-this-guide)
+2. [**Table of Contents**](#table-of-contents)
+3. [**Overview**](#overview)
+   1. [**What is EduMate?**](#what-is-edumate)
+   2. [**Understanding the Symbols and Colours**](#understanding-the-symbols-and-colours)
+   3. [**Glossary**](#glossary)
+4. [**Quick Start**](#quick-start)
+   1. [**System Requirements**](#system-requirements)
+   2. [**Installation Instructions**](#installation-instructions)
+5. [**User Interface**](#user-interface)
+   1. [**Person List**](#person-list)
+   2. [**Person Profile**](#person-profile)
+   3. [**Command Line**](#command-line)
+   4. [**Command Result**](#command-result)
+6. [**Commands**](#commands)
+   1. [**How to interpret the command format**](#how-to-interpret-the-command-format)
+   2. [**Arguments**](#arguments)
+      1. [**Name**](#name)
+      2. [**Phone Number**](#phone-number)
+      3. [**Email Address**](#email-address)
+      4. [**Home Address**](#home-address)
+      5. [**Telegram Handle**](#telegram-handle)
+      6. [**Group**](#group)
+      7. [**Module**](#module)
+   3. [**Basic Commands**](#basic-commands)
+      1. [**Exit the application `exit`**](#exit-the-application-exit)
+      2. [**Get help `help`**](#get-help-help)
+      3. [**List all contacts `list`**](#list-all-contacts-list)
+      4. [**Add a contact `add`**](#add-a-contact-add)
+      5. [**View a contact's profile `view`**](#view-a-contacts-profile-view)
+      6. [**Edit a contact's details `edit`**](#edit-a-contacts-details-edit)
+      7. [**Delete a contact `delete`**](#delete-a-contact-delete)
+      8. [**Add a label to a contact `tag`**](#tagging-a-module-to-an-existing-contact--tag)
+      9. [**Remove a label from a contact `tag`**](#untagging-a-module-from-an-existing-contact--untag)
+   4. [**Search Commands**](#search-commands)
```

Comment:
Would it perhaps be better to group search, storage, meet under advanced commands? It feels slightly inconsistent to have basic, advanced, and 3 independent commands.